rand_core/src/impls.rs

```diff
@@ -94,7 +94,7 @@ fn fill_via_chunks<T: Observable>(src: &[T], dest: &mut [u8]) -> (usize, usize)
         // This code is valid on all arches, but slower than the above:
         let mut i = 0;
         let mut iter = dest[..byte_len].chunks_exact_mut(size);
-        while let Some(chunk) = iter.next() {
+        for chunk in iter.by_ref() {
             chunk.copy_from_slice(src[i].to_le_bytes().as_ref());
```

Comment:
In such code I usually write `for chunk in &mut iter { .. }`, which I think Clippy accepts without issues.